aiounifi/models/voucher.py

```diff
@@ -0,0 +1,217 @@
+"""Hotspot vouchers as part of a UniFi network."""
+
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import NotRequired, Self, TypedDict
+
+from .api import ApiItem, ApiRequest
+
+
+class TypedVoucher(TypedDict):
+    """Voucher type definition."""
+
+    _id: str
+    site_id: str
+    note: str
+    code: int | str
+    quota: int
+    duration: float
+    qos_overwrite: bool
+    qos_usage_quota: NotRequired[str]
+    qos_rate_max_up: NotRequired[int]
+    qos_rate_max_down: NotRequired[int]
+    used: int
+    create_time: float
+    start_time: NotRequired[float]
+    end_time: NotRequired[float]
+    for_hotspot: bool
+    admin_name: str
+    status: str
+    status_expires: float
```

Comment:
The amount of raw.gets in the class properties are more than the amount of NotRequired here

Reply:
Sounds legit. I had tested much, but does not know if I have found all required and optional fields.
I will change it in one hour

Reply:
Now everything should fit. I had an issue with the "code" attribute in the fixture. The "code" only returns as a string.